modules/scala-support/src/main/scala/dev/guardrail/generators/scala/circe/CirceProtocolGenerator.scala

```diff
@@ -447,16 +449,25 @@ class CirceProtocolGenerator private (circeVersion: CirceModelGenerator, applyVa
       clsName: String,
       deps: List[scala.meta.Term.Name],
       encoder: Option[scala.meta.Defn.Val],
-      decoder: Option[scala.meta.Defn.Val]
+      decoder: Option[scala.meta.Defn.Val],
+      protocolParameters: List[ProtocolParameter[ScalaLanguage]]
   ) = {
     val extraImports: List[Import] = deps.map { term =>
       q"import ${term}._"
     }
+
+    val helperRegexTypes: List[Defn.Val] = protocolParameters.flatMap(_.propertyValidation).distinct.map { pattern: String =>
```

Comment:
Thread through Tracker, so you can use `.history` to give good error messages if `parse[Term].get` fails
```suggestion
    val helperRegexTypes: List[Defn.Val] = protocolParameters.flatTraverse(_.propertyValidation.indexedSequence.toList).map(_.distinct).map { pattern: Tracker[String] =>
```